### DISCUSSION TITLE
Preprocess and create distributed mesh to run large problems

### DISCUSSION THREAD

**Discussion by kks32** [cb-geo/mpm#711] [Ideas]
@cw646 I was wondering if we can preprocess an already distributed mesh and particle files. This way on each MPI task we only read the nodes and cells corresponding to the MPI task (and its ghost cells) and ignore the rest. This will reduce the memory consumption, however, we have to use `HALO_EXCHANGE` flag to do MPI calls. 

Pros
- Read large meshes and run bigger problems

Cons
- Can't do dynamic load balancing at this stage

Todo:
- What format should we have the distributed meshes and particles? maybe vtp?

**Comment by cw646** [comment]
I like this idea, dynamic load balancing already requires some work to fix - so it might be a good idea to do this modification before investing time into load balancing. 

Re the file type, vtp sounds good - unless there's anything that may be more memory efficient? h5?

**Reply by kks32** [reply]
We already write geometry.vtp file in mpm code and mesh.vtp, but that doesn't include the ghost cells. So we could use the existing codes was my though.
